cmd/server/main.go

```diff
@@ -124,44 +126,44 @@ func realMain(ctx context.Context) error {
 	// Install the CSRF protection middleware.
 	csrfController := csrfctl.New(ctx, h)
 	// TODO(mikehelmick) - there are many more configuration options for CSRF protection.
-	r.Use(csrf.Protect(config.CSRFAuthKey,
+	r.Use(middleware.ConfigureCSRF(ctx, config.CSRFAuthKey,
 		csrf.Secure(!config.DevMode),
 		csrf.SameSite(csrf.SameSiteLaxMode),
 		csrf.ErrorHandler(csrfController.HandleError()),
 	))
 
-	// Flash
-	// TODO(sethvargo): remove
-	r.Use(middleware.FlashHandler)
+	// Sessions
+	requireSession := middleware.RequireSession(ctx, sessions, h)
+	r.Use(requireSession)
 
 	// Create common middleware
-	requireAuth := middleware.RequireAuth(ctx, auth, db, h, sessions, config.SessionDuration)
-	requireAdmin := middleware.RequireRealmAdmin(ctx, h, sessions)
-	requireRealm := middleware.RequireRealm(ctx, h, sessions)
+	requireAuth := middleware.RequireAuth(ctx, auth, db, h, config.SessionDuration)
```

Comment:
Looking below, it looks like all sub controllers requireAuth, should you just install it on the root router?

Reply:
index and session controllers don't require auth